android/src/main/kotlin/com/airship/flutter/AirshipPlugin.kt

```diff
@@ -568,10 +587,135 @@ class AirshipPlugin : MethodCallHandler, FlutterPlugin {
 
     private fun openPreferenceCenter(call: MethodCall, result: Result) {
         val preferenceCenterID = call.arguments as String
-        PreferenceCenter.shared().open(preferenceCenterId = preferenceCenterID)
+
+        val sharedPref: SharedPreferences = context.getSharedPreferences("shared_preferences", Context.MODE_PRIVATE)
+        val enabled = sharedPref.getBoolean(AUTO_LAUNCH_PREFERENCE_CENTER_KEY, true)
+
+        if (enabled) {
+            PreferenceCenter.shared().open(preferenceCenterID)
+        } else {
+            EventManager.shared.notifyEvent(ShowPreferenceCenterEvent(preferenceCenterID))
+        }
         result.success(null)
     }
 
+    private fun getSubscriptionLists(call: MethodCall, result: Result) {
+
+        val subscriptionTypes = call.arguments as List<String>
+
+        val subscriptionLists = mutableMapOf<String, Any>()
+
+        scope.launch {
+            val r1 = UAirship.shared().contact.getSubscriptionLists(true)
+            suspendCoroutine<Unit> { continuation ->
+                r1.addResultCallback {
+                    if (it != null) {
+                        val map = mutableMapOf<String, Any>()
+                        for (list in it) {
+                            val values = list.value.toList()
+                            map[list.key] = values.map {it.toString()}
+                        }
+                        subscriptionLists["contact"] = map
+                    }
+                    continuation.resume(Unit)
+                }
+            }
+            val r2 = UAirship.shared().channel.getSubscriptionLists(true)
+            suspendCoroutine<Unit> { continuation ->
+                r2.addResultCallback {
+                    if (it != null) {
+                        subscriptionLists["channel"] = it.toList()
+                    }
+                    continuation.resume(Unit)
+                }
+            }
+            result.success(subscriptionLists)
+        }
+    }
+
+    private fun getPreferenceCenterConfig(call: MethodCall, result: Result) {
+        val preferenceCenterID = call.arguments as String
+        val pendingConfig = PreferenceCenter.shared().getConfig(preferenceCenterID)
+
+        pendingConfig.addResultCallback {
+            it?.let {
+                result.success(configDict(it))
+            }
+        }
+    }
+
+    private fun setAutoLaunchDefaultPreferenceCenter(call: MethodCall, result: Result) {
+        val enabled = call.arguments as Boolean
+
+        val sharedPref: SharedPreferences = context.getSharedPreferences("shared_preferences", Context.MODE_PRIVATE)
+        sharedPref.edit().putBoolean(AUTO_LAUNCH_PREFERENCE_CENTER_KEY, enabled)
+
+        result.success(null)
+    }
+
+    private fun configDict(config: PreferenceCenterConfig): Map<String, Any> {
```

Comment:
This could probably be shortened a bit using some of Kotlin's fancy collection transformation APIs, but I'm not gonna block on it. Might be worth adding SDK methods that return raw pref center json to make this easier to wrangle in the future, though?

Reply:
💯